python/text-to-speech/main.py

```diff
@@ -0,0 +1,292 @@
+
+    def validate_request(self, req: requests) -> None:
+        """
+        Validate the request data for Google text to speech.
+
+        Input:
+            req (request): The request provided by the user.
+
+        Raises:
+            ValueError: If any required value is missing or invalid.
+        """
+        if not req.variables.get("API_KEY"):
+            raise ValueError("Missing API_KEY.")
+        if not req.variables.get("PROJECT_ID"):
+            raise ValueError("Missing PROJECT_ID.")
+        self.api_key = req.variables.get("API_KEY")
+        self.project_id = req.variables.get("PROJECT_ID")
+
+    def speech(self, text: str, language: str) -> bytes:
+        """
+        Convert the given text into speech with the Google text to speech API.
+
+        Input:
+            text: The text to be converted into speech.
+            language: The language code (BCP-47 format).
+
+        Returns:
+            bytes: The synthezied speech in bytes.
+        """
+        # Instantiate a client.
+        client = texttospeech.TextToSpeechClient(
+            client_options={
+                "api_key": self.api_key,
+                "quota_project_id": self.project_id,
+            }
+        )
+        # Set the text input to be synthesized.
+        synthesis_input = texttospeech.SynthesisInput(text=text)
+        # Build the voice request, select the language code ("en-US")
+        # and the ssml voice gender is neutral.
+        voice = texttospeech.VoiceSelectionParams(
+            language_code=language,
+            ssml_gender=texttospeech.SsmlVoiceGender.NEUTRAL,
+        )
+        # Select the type of audio file you want returned.
+        audio_config = texttospeech.AudioConfig(
+            audio_encoding=texttospeech.AudioEncoding.MP3,
+        )
+        # Perform the text-to-speech request on the text input
+        # with the selected voice parameters and audio file type.
+        response = client.synthesize_speech(
+            input=synthesis_input,
+            voice=voice,
+            audio_config=audio_config,
+        )
+        return response.audio_content
+
+
+class Azure(TextToSpeech):
+    """Represent the implementation of Azure text to speech."""
+
+    VOICE = "en-US-ChristopherNeural"
+    GENDER = "Male"
+    REGION = "westus"
+
+    def validate_request(self, req: requests) -> None:
+        """
+        Validate the request data for Azure text to speech.
+
+        Input:
+            req (request): The request provided by the user.
+        Raises:
+            ValueError: If any required value is missing or invalid.
+        """
+        if not req.variables.get("API_KEY"):
+            raise ValueError("Missing API_KEY.")
+        self.api_key = req.variables.get("API_KEY")
+
+    def get_token(self, subscription_key: str) -> str:
+        """Return an Azure token for a given subscription key."""
+        fetch_token_url = (
```

Comment:
You could probably make this a class constant, similar to the REGION

Reply:
We fixed this.